test/e2e/ingress.go

```diff
@@ -243,31 +374,68 @@ var _ = Describe("GCE L7 LoadBalancer Controller", func() {
 		framework.beforeEach()
 		client = framework.Client
 		ns = framework.Namespace.Name
-		Expect(waitForRCPodsRunning(client, "kube-system", "glbc")).NotTo(HaveOccurred())
-		Expect(checkLeakedResources()).NotTo(HaveOccurred())
+		// Scaled down the existing Ingress controller so it doesn't interfere with the test.
+		Expect(scaleKubeSystemRC(client, lbContainerName, 0)).NotTo(HaveOccurred())
```

Comment:
I'm reading the pod manifest from the cluster/addons dir and replacing some bits for testing. In theory this is just the same as running with a test configuration (if you squint)?